File: sidecars/inflection-multi/inflection/domain/feature.py
from enum import Enum


class Feature(Enum):
    pass

    def json(self):
        return {"type": self.__class__.__name__.upper(), "value": self.name}


class Case(Feature):
    NOM = "Nom"
    GEN = "Gen"
    DAT = "Dat"
    ACC = "Acc"


class Number(Feature):
    SING = "Sing"
    PLUR = "Plur"


class Gender(Feature):
    MASC = "Masc"
    FEM = "Fem"
    NEUT = "Neut"


class Person(Feature):
    FIRST = "1"
    SECOND = "2"
    THIRD = "3"


class Tense(Feature):
    PAST = "Past"
    PRES = "Pres"
    FUT = "Fut"

File: grammr-ui/src/types/feature.ts
export interface Feature {
  type: string;
  value: string;
  fullIdentifier: string;
}

export enum FeatureType {
  CASE = 'Case',
  PERSON = 'Person',
  NUMBER = 'Number',
  TENSE = 'Tense',
  GENDER = 'Gender',
  MISC = 'Undefined',
}

export enum Case {
  NOM = 'Nominative',
  GEN = 'Genitive',
  DAT = 'Dative',
  ACC = 'Accusative',
  ABL = 'Ablative',
  VOC = 'Vocative',
  LOC = 'Locative',
  INS = 'Instrumental',
}

export enum Person {
  FIRST = 'First person',
  SECOND = 'Second person',
  THIRD = 'Third person',
}

export enum Number {
  SINGULAR = 'Singular',
  PLURAL = 'Plural',
}

export enum Tense {
  PAST = 'Past tense',
  IMP = 'Imperfect',
  PRES = 'Present tense',
  PQP = 'Plusquamperfect',
  FUT = 'Future tense',
}

export enum Gender {
  MASC = 'Masculine',
  FEM = 'Feminine',
  NEUT = 'Neuter',
}
